velox/functions/lib/LikeFunctionUtils.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LikeFunctionUtils.h"
+
+#include <optional>
+
+#include "velox/expression/EvalCtx.h"
```

Comment:
Please can you check if these header files are needed. I feel like many of them like Expr, ArrayBuilder are not needed.

Reply:
Thanks for pointing it out, they were used earlier but are no longer needed, removing all unnecessary headers